cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -891,16 +894,51 @@ void reader_impl::update_output_nullmasks_for_pruned_pages(cudf::host_span<bool
 
   std::for_each(
     page_and_mask_begin, page_and_mask_begin + pages.size(), [&](auto const& page_and_mask_pair) {
-      // Return early if the page is valid
+      // Return early if the page is valid - Note: dictionary pages are always valid
       if (thrust::get<1>(page_and_mask_pair)) { return; }
 
-      auto const& page     = thrust::get<0>(page_and_mask_pair);
-      auto const chunk_idx = page.chunk_idx;
-      auto const start_row = chunks[chunk_idx].start_row + page.chunk_row;
-      auto const end_row   = start_row + page.num_rows;
-      auto& input_col      = _input_columns[chunk_idx % num_columns];
-      auto max_depth       = input_col.nesting_depth();
-      auto* cols           = &_output_buffers;
+      // Get the current page
+      auto const& page = thrust::get<0>(page_and_mask_pair);
+
+      // Table row bounds
+      auto const table_start_row = skip_rows;
+      auto const table_end_row   = skip_rows + num_rows;
+
+      // Current page row bounds
+      auto const chunk_idx      = page.chunk_idx;
+      auto const page_start_row = chunks[chunk_idx].start_row + page.chunk_row;
+      auto const page_end_row   = page_start_row + page.num_rows;
+
+      // Check if the page is completely outside the table row bounds and return early if so
+      auto const is_page_out_of_bounds =
```

Comment:
In case we create multiple table chunks from a subpass pages, we must only use the pages that contribute to the current output table chunk